consensus/consensus-types/src/timeout_proposal.rs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{block::Block, common::Round};
+use libra_crypto::hash::{CryptoHash, CryptoHasher, HashValue, TimeoutProposalHasher};
+use serde::{Deserialize, Serialize};
+
+/// This structure contains all the information necessary to construct a signature
+/// on the equivalent of a timeout message
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct TimeoutProposal {
```

Comment:
The return of a timeout! There is some irony in that :).
I was trying to avoid having this struct because it's not really used anywhere as a separate entity. Note that you're planning to use it just as a container for passing the `round` and the `epoch`...

There are two approaches we could take:
1) No struct, pass the `round` and `epoch` for signing separately.
2) We have the struct, but then we also use it in the `TimeoutCertificate` (replace the separate fields for `round` and `epoch`). In this case I would probably call the struct `Timeout`, it's not just a proposal.

Reply:
I knew this would bring a smile to your face.

1. Personally I feel it is a little ambiguous what we are signing by passing around a round and an epoch, from this perspective this is a container to reduce errors.
2. Oh, yes, that is a good clean up. I will swap the fields in TimeoutCertificate to be a Timeout instead.